PluginCore/PluginCore/Helpers/ShortcutKeysConverter.cs

```diff
@@ -0,0 +1,224 @@
+            Add(Keys.NumPad5, "Num 5");
+            Add(Keys.NumPad6, "Num 6");
+            Add(Keys.NumPad7, "Num 7");
+            Add(Keys.NumPad8, "Num 8");
+            Add(Keys.NumPad9, "Num 9");
+            Add(Keys.Multiply, "Num *");
+            Add(Keys.Add, "Num +");
+            Add(Keys.Subtract, "Num -");
+            Add(Keys.Decimal, "Num .");
+            Add(Keys.Divide, "Num /");
+            Add(Keys.OemSemicolon, ";"); // Keys.Oem1
+            Add(Keys.Oemplus, "=");
+            Add(Keys.Oemcomma, ",");
+            Add(Keys.OemMinus, "-");
+            Add(Keys.OemPeriod, ".");
+            Add(Keys.OemQuestion, "/"); // Keys.Oem2
+            Add(Keys.Oemtilde, "~"); // Keys.Oem3
+            Add(Keys.OemOpenBrackets, "["); // Keys.Oem4
+            Add(Keys.OemPipe, "\\"); // Keys.Oem5
+            Add(Keys.OemCloseBrackets, "]"); // Keys.Oem6
+            Add(Keys.OemQuotes, "\""); // Keys.Oem7
+            Add(Keys.OemBackslash, "\\"); // Keys.Oem102
+            Add(Keys.Shift, "Shift");
+            Add(Keys.Control, "Ctrl");
+            Add(Keys.Alt, "Alt");
+        }
+
+        private static void Add(Keys key, string name)
+        {
+            names[key] = name;
+            keys[name] = key;
+        }
+
+        private static string ConvertToStringInternal(Keys keys)
+        {
+            // For performance reasons, instead of using a string or StringBuilder buffer and appending
+            // text such as Ctrl, Alt and Shift on it, use the string concatenation to utilize the compiler optimization,
+            // which turns them into string.Concat() calls.
+            if ((keys & Keys.Control) == Keys.Control)
+            {
+                if ((keys & Keys.Alt) == Keys.Alt)
+                {
+                    if ((keys & Keys.Shift) == Keys.Shift)
+                    {
+                        return Ctrl + Alt + Shift + GetString(keys & Keys.KeyCode);
+                    }
+                    return Ctrl + Alt + GetString(keys & Keys.KeyCode);
+                }
+                if ((keys & Keys.Shift) == Keys.Shift)
+                {
+                    return Ctrl + Shift + GetString(keys & Keys.KeyCode);
+                }
+                return Ctrl + GetString(keys & Keys.KeyCode);
+            }
+            if ((keys & Keys.Alt) == Keys.Alt)
+            {
+                if ((keys & Keys.Shift) == Keys.Shift)
+                {
+                    return Alt + Shift + GetString(keys & Keys.KeyCode);
+                }
+                return Alt + GetString(keys & Keys.KeyCode);
+            }
+            if ((keys & Keys.Shift) == Keys.Shift)
+            {
+                return Shift + GetString(keys & Keys.KeyCode);
+            }
+            return GetString(keys);
+        }
+
+        private static unsafe ShortcutKeys ConvertFromStringInternal(string value)
+        {
+            // For performance reasons, use an unsafe context and char pointers.
```

Comment:
Unsafe code pretty much increases performance by eliminating managed processes, like array bounds check, type safety, etc.
On a second thought, I think this code isn't that critical to use unsafe.

Scintilla control already uses unsafe code on almost all string operations, and the string class uses unsafe internally too. However codes that loop through chars in strings can have their performance improved by using unsafe without worrying about multi-bytes (such as this)
![image](https://cloud.githubusercontent.com/assets/13545633/15989621/b832b1f8-30b6-11e6-8a9c-77f5443cc43b.png)
If `ba` was `char*` instead of `string` (cool thing is that the syntax doesn't change) there wouldn't be any validity checks for the type of `ba`, checking if `ba` is `null`, index property accessing operation, or index boundary check. It would just be calculation of `ba + i + 1` and comparison to `'n'` etc.

I wouldn't worry about it too much, but I think it's not bad to consider it for better performance.